test/k8sT/Services.go

```diff
@@ -395,64 +456,74 @@ var _ = Describe("K8sServicesTest", func() {
 			})
 		})
 
+		// This label should be respected for all the service types, but testing for ClusterIP is enough.
+		// As service type does not influence if Cilium selects the service for management or not.
 		It("Checks service.kubernetes.io/service-proxy-name label implementation", func() {
 			serviceProxyLabelName := "service.kubernetes.io/service-proxy-name"
 
 			ciliumPods, err := kubectl.GetCiliumPods()
 			Expect(err).To(BeNil(), "Cannot get cilium pods")
 
-			clusterIP, _, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, echoServiceName)
-			Expect(err).Should(BeNil(), "Cannot get service %q ClusterIP", echoServiceName)
-			Expect(govalidator.IsIP(clusterIP)).Should(BeTrue(), "ClusterIP is not an IP")
-
-			By("Labelling echo service with dummy service-proxy-name")
-			res := kubectl.Exec(fmt.Sprintf("kubectl label services/%s %s=%s", echoServiceName, serviceProxyLabelName, "dummy-lb"))
-			res.ExpectSuccess("cannot label service")
-
-			// Wait for all cilium pods to remove the service from its list.
-			By("Validating that Cilium is not handling the service")
-			Eventually(func() int {
-				validPods := 0
-				for _, pod := range ciliumPods {
-					serviceRes := kubectl.CiliumExecMustSucceed(
-						context.TODO(), pod, "cilium service list", "Cannot retrieve services on cilium Pod")
-
-					if !strings.Contains(serviceRes.Stdout(), clusterIP) {
-						validPods++
+			serviceNames := []string{echoServiceName}
+			// TODO: Uncomment after https://github.com/cilium/cilium/pull/14138 is merged
+			// if dualStackSupportEnabled {
```

Comment:
This looks odd, we can't "know" the order the variable will be initialized so it would be better executing `helpers.DualStackSupported()` directly here.